### ISSUE TITLE
Saving changes causes the ViewContent to be reloaded

### ISSUE THREAD

**Issue by alanmcgovern** [mono/monodevelop#3951]
1) Open a storyboard (or any file?).
2) Make a modification
3) Save the changes

Result:
Document.Reload is invoked. In turn this invokes ViewContent.Load, which is a destructive operation as it initializes a ViewContent, and in our case will result in the undo/redo stack being thrown away because it's initialized in that method along with the design surface etc.

This is a change in VSM behaviour and I believe it's a bug. The stacktrace for this is:
```
MonoDevelop.Ide.Gui.Document.Reload() in /Users/alan/Projects/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Document.cs:369
MonoDevelop.Ide.Gui.DocumentRegistry.HandleFileServiceChange(object sender, MonoDevelop.Core.FileEventArgs e) in /Users/alan/Projects/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DocumentRegistry.cs:81
MonoDevelop.Core.EventQueue.EventData<MonoDevelop.Core.FileEventArgs>.Invoke() in /Users/alan/Projects/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Core/FileService.cs:860
MonoDevelop.Core.EventQueue.Thaw() in /Users/alan/Projects/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Core/FileService.cs:930
MonoDevelop.Core.FileService.ThawEvents() in /Users/alan/Projects/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Core/FileService.cs:719
MonoDevelop.Ide.Gui.Document.SaveTask() in /Users/alan/Projects/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Document.cs:441
System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start<MonoDevelop.Ide.Gui.Document.<SaveTask>d__71>(MonoDevelop.Ide.Gui.Document stateMachine) in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:316
MonoDevelop.Ide.Gui.Document.SaveTask() in 
MonoDevelop.Ide.Gui.Document.(System.Threading.Tasks.UnwrapPromise<System.Threading.Tasks.TaskExtensions.VoidResult> t) in /Users/alan/Projects/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Document.cs:364
System.Threading.Tasks.ContinuationResultTaskFromTask<System.Threading.Tasks.Task>.InnerInvoke() in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/Tasks/TaskContinuation.cs:111
System.Threading.Tasks.Task.Execute() in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/Tasks/Task.cs:2509
System.Threading.Tasks.Task.ExecutionContextCallback(System.Threading.Tasks.ContinuationResultTaskFromTask<System.Threading.Tasks.Task> obj) in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/Tasks/Task.cs:2872
System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Threading.Tasks.ContinuationResultTaskFromTask<System.Threading.Tasks.Task> state, bool preserveSyncCtx) in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:957
System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Threading.Tasks.ContinuationResultTaskFromTask<System.Threading.Tasks.Task> state, bool preserveSyncCtx) in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:904
System.Threading.Tasks.Task.ExecuteWithThreadLocal(System.Threading.Tasks.ContinuationResultTaskFromTask<System.Threading.Tasks.Task> currentTaskSlot) in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/Tasks/Task.cs:2834
System.Threading.Tasks.Task.ExecuteEntry(bool bPreventDoubleExecution) in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/Tasks/Task.cs:2767
System.Threading.Tasks.SynchronizationContextTaskScheduler.PostCallback(System.Threading.Tasks.ContinuationResultTaskFromTask<System.Threading.Tasks.Task> obj) in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/threading/Tasks/TaskScheduler.cs:728
MonoDevelop.Ide.DispatchService.GtkSynchronizationContext.TimeoutProxy.HandlerInternal(System.IntPtr data) in /Users/alan/Projects/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/DispatchService.cs:77
Gtk.Application.gtk_main() in 
Gtk.Application.Run() in /Users/builder/jenkins/workspace/build-package-osx-mono/2017-12/external/bockbuild/builds/gtk-sharp-None/gtk/Application.cs:145
MonoDevelop.Ide.IdeApp.Run() in /Users/alan/Projects/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/Ide.cs:404
MonoDevelop.Ide.IdeStartup.Run(MonoDevelop.Ide.MonoDevelopOptions options) in /Users/alan/Projects/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs:300
MonoDevelop.Ide.IdeStartup.Main(string[] args, Xamarin.IdeCustomization.XamarinCustomizer customizer) in /Users/alan/Projects/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs:619
MonoDevelop.Startup.MonoDevelopMain.Main(string[] args) in /Users/alan/Projects/monodevelop/main/src/core/MonoDevelop.Startup/MonoDevelop.Startup/MonoDevelopMain.cs:39```

> VS bug #572337

**Comment by alanmcgovern** [comment]
NOTE: We should adjust our code to check if `FileOpenInformation.IsReloadOperation`  is true/false to avoid destructive operations in the event of a proper reload, but we should not be reloading in this case either way.

**Comment by slluis** [comment]
Is this on master or 15.6?

**Comment by alanmcgovern** [comment]
This is a blocker for any 15.7 preview!

**Comment by alanmcgovern** [comment]
💯 

Thanks!

I'll test this out shortly!

**Comment by alanmcgovern** [comment]
Also what's the procedure for ensuring things marked as '15.7' are actually backported? The issue is closed whenever the *master* pr merges, but we want it to close when master and 15.7 have the patch.

Should we be creating duplicate issues for everything, or just asking for the backports to be created before merging the master commit so we don't end up closing the issue before the backport is created?

**Comment by slluis** [comment]
If an issue has a target milestone, it will only be automatically closed when the fix is merged to the corresponding release branch.

**Comment by alanmcgovern** [comment]
Oh perfect, thanks!

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-24] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #4046 merged by: Mike Kr??ger  
 Author: mono  
 Commit: 9326b144c432cbeeb986addf571fe24d858f5da0 (xamarin/monodevelop)
